<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:macro name="odrive_drive_ros2_control" params="name ros2_control_hardware_type">

        <ros2_control name="${name}" type="system">
            <hardware>
              <xacro:if value="${ros2_control_hardware_type == 'main'}">
                <plugin>odrive_ros2_control_plugin/ODriveHardwareInterface</plugin>
                <param name="can">can0</param>
                <!-- <plugin>mock_components/GenericSystem</plugin> -->
              </xacro:if>
              <xacro:if value="${ros2_control_hardware_type == 'sim'}">
                <plugin>mock_components/GenericSystem</plugin>
              </xacro:if>
            </hardware>
            <joint name="front_right">
                <param name="node_id">0</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="front_left">
                <param name="node_id">1</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="middle_right">
                <param name="node_id">2</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="middle_left">
                <param name="node_id">3</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="back_right">
                <param name="node_id">4</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
            <joint name="back_left">
                <param name="node_id">5</param>
                <param name="gear_ratio">50.0</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>

        </ros2_control>
    </xacro:macro>
</robot>
